ir/memory.cpp

```diff
@@ -172,17 +172,18 @@ void Pointer::is_disjoint(const expr &len1, const Pointer &ptr2,
                            len2.zextOrTrunc(m.bits_size_t)));
 }
 
-expr Pointer::is_block_alive() {
+expr Pointer::is_block_alive() const {
   return m.blocks_liveness.load(get_bid());
 }
 
-expr Pointer::is_at_heap() {
+expr Pointer::is_at_heap() const {
   return m.blocks_kind.load(get_bid()) == 1;
 }
 
 ostream& operator<<(ostream &os, const Pointer &p) {
-  os << "pointer(" << (p.is_local().simplify().isTrue() ? "local" : "non-local")
-     << ", block_id=";
+  os << "pointer(" << (p.is_local().simplify().isTrue() ? "local" : "non-local");
+  os << ", " << (p.is_at_heap().simplify().isTrue() ? "heap" : "non-heap");
```

Comment:
uhm, I don't think this will work. When a pointer is constructed for printing, the smt expressions are constant (taken from the model). The memory isn't filled with the model. Anyway, this is a property of the memory block, not of the pointer /per se/.
My suggestion is to revert this change. We can (should) add a printer to the memory later, showing the properties of each block (size, type, etc) and contents.

Reply:
This worked for input cases in alive-tv test, printing such as
```
Source:                                                                                                      
* %ptr = pointer(local, heap, block_id=256, offset=0)
```
or 
```
Example:                                                                                                     
* %ptr0 = pointer(non-local, non-heap, block_id=0, offset=3458764513820540927) 
```
so I guessed there was magic in simplify() thingy.
But I am not very sure, and also fine with printing them at memory side, so I reverted the change.